### PR TITLE
[DomCrawler] Added argument `$default` to method `Crawler::attr()`

### PR DESCRIPTION
**Pull request by Rastishka**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | no
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

Added argument `$default` to method `->attr()` like `->text()` and `->html()` have.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Do we really need this? My feeling is that we now have the `??` operator and it's nicer to use.

**Comment by antonkomarev** [comment]
I'd prefer to use native `??`. It feels more obvious about the result.

**Comment by Rastishka** [comment]
> Do we really need this? My feeling is that we now have the `??` operator and it's nicer to use.

Yes we need, because setting default value also prevents throwing exception when node does not exist (methods text() and html() act similar).

So this code:
```
try {
    $value = $dom->filter('a.someclass')->attr('href');
} catch (\InvalidArgumentException $e) {
    $value = '';
}
```
can be simplifies to
` $value = $dom->filter('a.someclass')->attr('href', '');`

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -522,17 +522,23 @@ public function children(string $selector = null): static
...
      * Returns the attribute value of the first node of the list.
      *
+     * @param string|null $default When not null: the value to return when the node or attribute is empty
+     *
      * @throws \InvalidArgumentException When current node is empty
      */
-    public function attr(string $attribute): ?string
+    public function attr(string $attribute, string $default = null): ?string
     {
         if (!$this->nodes) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DomCrawler/Crawler.php]: To keep BC:
```suggestion
    public function attr(string $attribute/* , string $default = null */): ?string
    {
        $default = \func_num_args() > 1 ? func_get_arg(1) : null;
        if (!$this->nodes) {
```

**Review by nicolas-grekas — COMMENTED** [review]
Thanks for the explanation. Works for me.

**Comment by nicolas-grekas** [comment]
Please also update the UPGRADE-6.4.md file, and please rebase to get rid of merge commits (PRs shouldn't contain any).

**Comment by nicolas-grekas** [comment]
Thank you @Rastishka.

**Pull request merged**